### PR TITLE
#3145: Update admin content listing to show scheduled status

### PR DESCRIPTION
**Pull request by laryn**

### CODE REVIEW

**Comment by herbdool** [comment]
@laryn looks good! Should we use a different date format that's more human friendly?

File: core/modules/node/config/views.view.node_admin_content.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "name": "node_admin_content",
     "description": "Administrative listing for managing content.",
     "module": "node",
-    "storage": 4,
+    "storage": 2,
```
Comment (herbdool) [inline core/modules/node/config/views.view.node_admin_content.json]: I think it needs to stay as 4. 2 means it's no longer default I believe.

**Review by herbdool — COMMENTED** [review]
We'll also need an update hook to update everyone's view. But only do so if they haven't customized it since we don't want to remove them.

File: core/modules/node/config/views.view.node_admin_content.json (hunk 1 above)
Comment (laryn) [reply core/modules/node/config/views.view.node_admin_content.json]: Good catch.

**Comment by klonos** [comment]
@laryn I noticed that the view in the sandbox has `Unpublished<br /><span class="breadcrumb">(Draft)</span>` for No results text, and also `Unpublished<br /><span class="breadcrumb">(Scheduled for [scheduled])</span>` for rewrite text...

a) I don't see any of that exported in the config changes in the PR
b) not sure about `class="breadcrumb"`

**Comment by laryn** [comment]
@klonos All valid critiques for sure. This PR was intended more as a proof of concept than a final PR as I was/am assuming there might be feedback in the issue as to whether this approach was a good one. If we get a thumbs up on the direction I can clean it up. The latest changes were made directly in the sandbox on the fly and I haven't updated the PR yet.

**Comment by laryn** [comment]
Reboot sandbox.

**Comment by herbdool** [comment]
@laryn in my example I had "Not published (scheduled)". What do you think of that? Instead of just "Not published". I had used your initial view as a starting point

**Comment by laryn** [comment]
@herbdool Yes, makes sense. I'll revise.

**Comment by herbdool** [comment]
@laryn sorry one more thing: can you make Scheduled column sortable in the table? Then I think this is ready to merge!

**Comment by laryn** [comment]
Another good point!

File: core/modules/node/node.install [hunk 2]
```diff
@@ -1173,6 +1173,23 @@ function node_update_1016() {
...
+  // See if the content admin view has been modified on this site.
+  $config = config('views.view.node_admin_content');
+
+  // Only update if view isn't overridden
+  if ($config->get('storage') != VIEWS_STORAGE_OVERRIDE) {
+    $path = backdrop_get_path('module', 'node') . '/config/';
+    $contents = file_get_contents($path . 'views.view.node_admin_content.json');
+    $data = json_decode($contents, true);
+    $config->setData($data);
+    $config->save();
```
Comment (quicksketch) [inline core/modules/node/node.install]: I don't think we can just pull in the default configuration like this. Currently we have to make these changes explicitly by duplicating the changed config into the update hook. See https://github.com/backdrop/backdrop-issues/issues/3347.

Comment (herbdool) [reply core/modules/node/node.install]: @quicksketch I added a commit via github to convert json to php array, but now have a conflict. I haven't tried fixing another person's branch so got to take a careful look on how.

**Comment by herbdool** [comment]
I don't know best way to resolve this conflict without merging right away. So I just created a new PR https://github.com/backdrop/backdrop/pull/2450. It is the same patch as this one, but have merged in upstream changes.

**Comment by quicksketch** [comment]
Replaced by https://github.com/backdrop/backdrop/pull/2450.

**Pull request closed without merging**